protos/flyteidl/plugins/sagemaker/hyperparameter_tuning_job.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+package flyteidl.plugins.sagemaker;
+
+option go_package = "github.com/lyft/flyteidl/gen/pb-go/flyteidl/plugins";
+
+import "flyteidl/plugins/sagemaker/parameter_ranges.proto";
+import "flyteidl/plugins/sagemaker/training_job.proto";
+
+// The hyperparameter tuning job
+message HyperparameterTuningJob {
+    // The underlying training job that the hyperparameter tuning job will launch during the process
+    TrainingJob training_job = 1;  // TODO: This hierarchical definition is tricky!
+
+    // The maximum number of training jobs that an hpo job can launch. For resource limit purpose.
+    int64 max_number_of_training_jobs = 2;
+
+    // The maximum number of concurrent training job that an hpo job can launch
+    int64 max_parallel_training_jobs = 3;
+}
+
+// The objective of the hyperparameter tuning
+message HyperparameterTuningObjective {
+    enum HyperparameterTuningObjectiveType {
```

Comment:
I think this should be just called `Type` instead.. otherwise you will reference it by the very long name: `HyperparameterTuningObjective.HyperparameterTuningObjectiveType.MINIMIZE` ... etc.

Reply:
plz double check my claim in the generated python or by looking at other enums we have already added :)

Reply:
You are right it is really long and that bothered me lol